velox/functions/lib/Re2Functions.cpp

```diff
@@ -698,6 +698,88 @@ class Re2ExtractAll final : public VectorFunction {
   }
 };
 
+void re2SplitAll(
+    exec::VectorWriter<Array<Varchar>>& resultWriter,
+    const RE2& re,
+    const exec::LocalDecodedVector& inputStrs,
+    const int row,
+    std::vector<re2::StringPiece>& groups) {
+  resultWriter.setOffset(row);
+
+  auto& arrayWriter = resultWriter.current();
+
+  const StringView str = inputStrs->valueAt<StringView>(row);
+  const re2::StringPiece input = toStringPiece(str);
+  size_t pos = 0;
+
+  while (
+      re.Match(input, pos, input.size(), RE2::UNANCHORED, groups.data(), 1)) {
+    const re2::StringPiece fullMatch = groups[0];
+    const re2::StringPiece subMatch =
+        input.substr(pos, fullMatch.data() - input.data() - pos);
+
+    arrayWriter.add_item().setNoCopy(
+        StringView(subMatch.data(), subMatch.size()));
+    pos = fullMatch.data() + fullMatch.size() - input.data();
+    if (UNLIKELY(fullMatch.size() == 0)) {
+      ++pos;
+    }
+  }
+
+  if (pos < input.size()) {
+    const re2::StringPiece remaining = input.substr(pos);
+    arrayWriter.add_item().setNoCopy(
+        StringView(remaining.data(), remaining.size()));
+  } else if (pos == input.size()) {
+    arrayWriter.add_item().setNoCopy(StringView(nullptr, 0));
+  }
+
+  resultWriter.commit();
+}
+
+class Re2SplitAllConstantPattern final : public VectorFunction {
+ public:
+  explicit Re2SplitAllConstantPattern(StringView pattern)
+      : re_(toStringPiece(pattern), RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      EvalCtx& context,
+      VectorPtr& resultRef) const final {
+    try {
+      checkForBadPattern(re_);
```

Comment:
I think it is better to move this check to `makeRe2SplitAll` to get bad pattern checked one time.

Reply:
Thanks, I understand your means, but I think may it is better to keep it up now, because there is `context` in `apply` method, it can record exception for bad pattern. 
If move `checkForBadPattern` to `makeRe2SplitAll`,  maybe unable to find problems conveniently

Reply:
It looks not a strong reason. As the pattern is always constant, I think it may be not necessary to do this check for each incoming batch. cc @mbasmanova

Reply:
If pattern is constant, then let's make sure to check it once during function creation and use AlwaysFailingVectorFunction.